addons/stock/models/stock_quant.py

```diff
@@ -110,6 +114,7 @@ def _domain_product_id(self):
     inventory_date = fields.Date(
         'Scheduled Date', compute='_compute_inventory_date', store=True, readonly=False,
         help="Next date the On Hand Quantity should be counted.")
+    last_count_date = fields.Date(help='Last time the Quantity was Updated')
```

Comment:
I would rather do a compute base on `stock.move.line` with inventory flag. When the `stock.quant` are empty, they are deleted. So if the user empty a location and refill it after we will lose the data